File: omnipod-dash/src/test/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/response/SetUniqueIdResponseTest.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response;

import org.apache.commons.codec.DecoderException;
import org.apache.commons.codec.binary.Hex;
import org.junit.Test;

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition.PodStatus;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotSame;

public class SetUniqueIdResponseTest {
    @Test
    public void testValidResponse() throws DecoderException {
        byte[] encoded = Hex.decodeHex("011B13881008340A50040A00010300040308146CC1000954D402420001");
        SetUniqueIdResponse response = new SetUniqueIdResponse(encoded);

        assertArrayEquals(encoded, response.getEncoded());
        assertNotSame(encoded, response.getEncoded());
        assertEquals(ResponseType.ACTIVATION_RESPONSE, response.getResponseType());
        assertEquals(ResponseType.ActivationResponseType.SET_UNIQUE_ID_RESPONSE, response.getActivationResponseType());

        assertEquals(ResponseType.ACTIVATION_RESPONSE.getValue(), response.getMessageType());
        assertEquals(27, response.getMessageLength());
        assertEquals(5000, response.getPulseVolumeInTenThousandthMicroLiter());
        assertEquals(16, response.getDeliveryRate());
        assertEquals(8, response.getPrimeRate());
        assertEquals(52, response.getNumberOfEngagingClutchDrivePulses());
        assertEquals(10, response.getNumberOfPrimePulses());
        assertEquals(80, response.getPodExpirationTimeInHours());
        assertEquals(4, response.getFirmwareVersionMajor());
        assertEquals(10, response.getFirmwareVersionMinor());
        assertEquals(0, response.getFirmwareVersionInterim());
        assertEquals(1, response.getBleVersionMajor());
        assertEquals(3, response.getBleVersionMinor());
        assertEquals(0, response.getBleVersionInterim());
        assertEquals(4, response.getProductId());
        assertEquals(PodStatus.UID_SET, response.getPodStatus());
        assertEquals(135556289L, response.getLotNumber());
        assertEquals(611540L, response.getPodSequenceNumber());
        assertEquals(37879809L, response.getUniqueIdReceivedInCommand());
    }
}

File: omnipod-dash/src/test/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/response/SetUniqueIdResponseTest.kt
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition.PodStatus
import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response.ResponseType
import org.apache.commons.codec.DecoderException
import org.apache.commons.codec.binary.Hex
import org.junit.Assert
import org.junit.Test

class SetUniqueIdResponseTest {

    @Test @Throws(DecoderException::class) fun testValidResponse() {
        val encoded = Hex.decodeHex("011B13881008340A50040A00010300040308146CC1000954D402420001")
        val response = SetUniqueIdResponse(encoded)

        Assert.assertArrayEquals(encoded, response.encoded)
        Assert.assertNotSame(encoded, response.encoded)
        Assert.assertEquals(ResponseType.ACTIVATION_RESPONSE, response.responseType)
        Assert.assertEquals(ResponseType.ActivationResponseType.SET_UNIQUE_ID_RESPONSE, response.activationResponseType)
        Assert.assertEquals(ResponseType.ACTIVATION_RESPONSE.value, response.getMessageType())
        Assert.assertEquals(27.toShort(), response.getMessageLength())
        Assert.assertEquals(5000.toShort(), response.getPulseVolumeInTenThousandthMicroLiter())
        Assert.assertEquals(16.toShort(), response.getDeliveryRate())
        Assert.assertEquals(8.toShort(), response.getPrimeRate())
        Assert.assertEquals(52.toShort(), response.getNumberOfEngagingClutchDrivePulses())
        Assert.assertEquals(10.toShort(), response.getNumberOfPrimePulses())
        Assert.assertEquals(80.toShort(), response.getPodExpirationTimeInHours())
        Assert.assertEquals(4.toShort(), response.getFirmwareVersionMajor())
        Assert.assertEquals(10.toShort(), response.getFirmwareVersionMinor())
        Assert.assertEquals(0.toShort(), response.getFirmwareVersionInterim())
        Assert.assertEquals(1.toShort(), response.getBleVersionMajor())
        Assert.assertEquals(3.toShort(), response.getBleVersionMinor())
        Assert.assertEquals(0.toShort(), response.getBleVersionInterim())
        Assert.assertEquals(4.toShort(), response.getProductId())
        Assert.assertEquals(PodStatus.UID_SET, response.getPodStatus())
        Assert.assertEquals(135556289L, response.getLotNumber())
        Assert.assertEquals(611540L, response.getPodSequenceNumber())
        Assert.assertEquals(37879809L, response.getUniqueIdReceivedInCommand())
    }
}